src/emqx_router.erl

```diff
@@ -266,28 +266,34 @@ maybe_trans(Fun, Args) ->
                   end, [])
     end.
 
+%% The created fun only terminates with explicit exception
+-dialyzer({nowarn_function, [trans/2]}).
+
 -spec(trans(function(), list(any())) -> ok | {error, term()}).
 trans(Fun, Args) ->
-    %% trigger selective receive optimization of compiler,
-    %% ideal for handling bursty traffic.
-    Ref = erlang:make_ref(),
-    Owner = self(),
-    {WPid, RefMon} = spawn_monitor(
-                       fun() ->
-                               Res = case mnesia:transaction(Fun, Args) of
-                                         {atomic, Ok} -> Ok;
-                                         {aborted, Reason} -> {error, Reason}
-                                     end,
-                               Owner ! {Ref, Res}
-                       end),
+    {WPid, RefMon} =
+        spawn_monitor(
+            fun() ->
```

Comment:
put some comments that this `fun` assumes it will not exit unexpectedly because Fun runs in mnesia trans, future changes should prevent adding function calls that might exit the process which will return something unexpected to the parent.

Reply:
done